### PR TITLE
add -o option to choose XML output folder

### PR DESCRIPTION
**Pull request by n-a-t-e**
Just adding an option to put the generated XML into a specified folder. The other stuff in this PR are just minor changes to satisfy my linter. The `-o` is optional

eg:

`python -m metadata_xml -f sample_records/record.yml -o ./xml`

### CODE REVIEW

File: metadata_xml/template_functions.py [hunk 1]
```diff
@@ -33,25 +42,31 @@ def normalize_datestring(datestring, format_='default'):
...
-    def list_keys_in_dict(d, all_keys=[]):
-        # recursive function, works on nested dict for example
-        for k, v in d.items():
-            if isinstance(v, dict):
-                all_keys.append(k)
-                list_keys_in_dict(v, all_keys)
+    '''Lists all languages used, so that we can list all
+    languages used in the otherLocale section of XML'''
+    def list_keys_in_record(record_subset, all_keys=[]):
+        'recursive function, works on nested dict for example'
```
Comment (jeffcullis) [inline metadata_xml/template_functions.py]: Minor but docstrings should usually be triple double quotes per PEP:
https://www.python.org/dev/peps/pep-0257/

Comment (n-a-t-e) [reply metadata_xml/template_functions.py]: fixed thx

File: metadata_xml/__main__.py [hunk 2]
```diff
@@ -9,22 +9,29 @@
...
+import os
 import argparse
 import yaml
-import os
+from metadata_xml.template_functions import metadata_to_xml
 
-if __name__ == '__main__':
 
+def main():
+    'Handes argparse and calls metadata_to_xml()'
```
Comment (jeffcullis) [inline metadata_xml/__main__.py]: typo: handles

**Review by jeffcullis — APPROVED** [review]
A couple minor formatting issues that can be easily fixed, otherwise approved.

**Pull request merged**